secant/Mocked Dependencies/OnboardingMocks.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  OnboardingMocks.swift
+//  secantTests
+//
+//  Created by Francisco Gindre on 9/20/21.
+//
+
+import Foundation
+class OnboardingStepProviderBuilder {
+    var steps: [OnboardingStep] = []
+    var startingAt: Int = 0
+
+    func add(_ step: OnboardingStep) -> Self {
```

Comment:
Does a builder with an add function help anything here vs just directly instantiating a SequencedOnboardingStepProvider?

Reply:
builders look good on tests. just preference. no specific tradeoff.

Reply:
I'd say if its just for looks, this would go against the typical swiftly flow of just using value types and instantiating directly.